tests/test_osfs.py

```diff
@@ -94,3 +94,19 @@ def test_unicode_paths(self):
                 self.assertTrue(os.path.isdir(fs_dir))
         finally:
             shutil.rmtree(dir_path)
+
+
+class TestOSFS(OSFSTestBase, unittest.TestCase):
+    pass
+
+
+class TestOSFSScandir(OSFSTestBase, unittest.TestCase):
+
+    def test_scandir_purepython(self):
+        with mock.patch.object(self.fs, '_scandir', self.fs._scandir_py):
```

Comment:
I think I see why you have the 2 functions defined now. I think you can modif `tox.ini` file to have separate environments for with and without the scandir module. Otherwise its not going to include this in coverage.